feature/policy_forwarding/encapsulation/otg_tests/encap_gre_ipv4/README.md

```diff
@@ -0,0 +1,88 @@
+
+    ```
+                             |         | --eBGP-- | ATE Port 2 |
+        [ ATE Port 1 ] ----  |   DUT   |          |            |
+                             |         | --eBGP-- | ATE Port 3 |
+    ```
+
+*   Traffic is generated from ATE Port 1.
+*   ATE Port 2 is used as the destination port for encapsulated 
+    traffic.
+*   ATE Port 3 is used as the fallback destination for
+    pass-through traffic.
+
+### Configuration
+
+1.  All DUT Ports are configured as a singleton IP interfaces.
+
+2.  IPv4 and IPv6 static routes to test destination networks IPV4-DST/IPV6-DST are configured on DUT towards ATE Port 3.
+
+3.  Another set of IPv4 static routes to 32x IPv4 GRE encap destinations towards ATE Port 2.
+
+4.  2 IPv4 and 2 IPv6 source prefixes will be used to generate tests traffic 
+(SRC1-SRC2). Apply policy-forwarding with 4 rules to DUT Port 1:
+    - Match IPV4-SRC1 and accept/foward.
+    - Match IPV6-SRC1 and accept/foward.
+    - Match IPV4-SRC2 and encapsulate to 32 IPv4 GRE destinations.
+    - Match IPV6-SRC2 and encapsulate to 32 IPv4 GRE destinations.
+
+## Test cases
+
+### PF-1.1.1: Verify PF GRE encapsulate action for IPv4 traffic
+Generate traffic on ATE Port 1 from IPV4-SRC2 from a random combination of 1000 source addresses to IPV4-DST.
+
+Verify:
+
+*  All traffic received on ATE Port 2 GRE-encapsulated.
+*  No packet loss when forwarding.
+*  Traffic equally load-balanced across 32 GRE destinations
+*  Verify PF packet counters matching traffic generated.
+
+### PF-1.1.2: Verify PF GRE encapsulate action for IPv6 traffic
+Generate traffic on ATE Port 1 from IPV6-SRC2 from a random combination of 1000 source addresses to IPV6-DST.
+
+Verify:
+
+*  All traffic received on ATE Port 2 GRE-encapsulated.
+*  No packet loss when forwarding.
+*  Traffic equally load-balanced across 32 GRE destinations
+*  Verify PF packet counters matching traffic generated.
+
+### PF-1.1.3: Verify PF IPV4 forward action
+Generate traffic on ATE Port 1 from sources IPV4-SRC1 to IPV4-DST.
+
+Verify:
+
+*  All traffic received on ATE Port 3.
+*  No packet loss when forwarding.
+
+### PF-1.1.4: Verify PF IPV6 forward action
+Generate traffic on ATE Port 1 from sources IPV6-SRC1 to IPV6-DST.
+
+Verify:
+
+*  All traffic received on ATE Port 3.
+*  No packet loss when forwarding.
+
+
+## Config Parameter Coverage
+
+*   `/network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/ipv4/config/source-address`
+*   `/network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/action/encapsulate-gre`
+*   `/network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/action/encapsulate-gre/targets/target/config/destinations`
+
+
+## Telemetry Parameter Coverage
+
+*   `/network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/state/matched-pkts`
+*   `/network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/state/matched-octets`
```

Comment:
Probably you need the match rule?  Also there are some other gre encap related paths, do you want to cover any of these (ok if you don't but since the basic match rules where missing, I thought I would list these for you to consider):

For example the match rule is: 

[/network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/action/encapsulate-gre/config/identifying-prefix](https://openconfig.net/projects/models/schemadocs/yangdoc/openconfig-network-instance.html#network-instances-network-instance-policy-forwarding-policies-policy-rules-rule-action-encapsulate-gre-config-identifying-prefix)

Here's a dump from https://openconfig.net/projects/models/paths/ matching the search string: "policy forwarding gre config/"




openconfig-network-instance | /network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/action/config/decapsulate-gre | leaf | boolean
-- | -- | -- | --
openconfig-network-instance | /network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/action/encapsulate-gre/config/identifying-prefix | leaf | ip-prefix
openconfig-network-instance | /network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/action/encapsulate-gre/targets/target/config/id | leaf | string
openconfig-network-instance | /network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/action/encapsulate-gre/targets/target/config/source | leaf | ip-address
openconfig-network-instance | /network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/action/encapsulate-gre/targets/target/config/destination | leaf | ip-prefix
openconfig-network-instance | /network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/action/encapsulate-gre/targets/target/config/ip-ttl | leaf | uint8


openconfig-network-instance	[/network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/action/config/decapsulate-gre](https://openconfig.net/projects/models/schemadocs/yangdoc/openconfig-network-instance.html#network-instances-network-instance-policy-forwarding-policies-policy-rules-rule-action-config-decapsulate-gre)	leaf	boolean
openconfig-network-instance	[/network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/action/encapsulate-gre/config/identifying-prefix](https://openconfig.net/projects/models/schemadocs/yangdoc/openconfig-network-instance.html#network-instances-network-instance-policy-forwarding-policies-policy-rules-rule-action-encapsulate-gre-config-identifying-prefix)	leaf	ip-prefix
openconfig-network-instance	[/network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/action/encapsulate-gre/targets/target/config/id](https://openconfig.net/projects/models/schemadocs/yangdoc/openconfig-network-instance.html#network-instances-network-instance-policy-forwarding-policies-policy-rules-rule-action-encapsulate-gre-targets-target-config-id)	leaf	string
openconfig-network-instance	[/network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/action/encapsulate-gre/targets/target/config/source](https://openconfig.net/projects/models/schemadocs/yangdoc/openconfig-network-instance.html#network-instances-network-instance-policy-forwarding-policies-policy-rules-rule-action-encapsulate-gre-targets-target-config-source)	leaf	ip-address
openconfig-network-instance	[/network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/action/encapsulate-gre/targets/target/config/destination](https://openconfig.net/projects/models/schemadocs/yangdoc/openconfig-network-instance.html#network-instances-network-instance-policy-forwarding-policies-policy-rules-rule-action-encapsulate-gre-targets-target-config-destination)	leaf	ip-prefix
openconfig-network-instance	[/network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/action/encapsulate-gre/targets/target/config/ip-ttl](https://openconfig.net/projects/models/schemadocs/yangdoc/openconfig-network-instance.html#network-instances-network-instance-policy-forwarding-policies-policy-rules-rule-action-encapsulate-gre-targets-target-config-ip-ttl)	leaf	uint8